pkg/florist/netstat.go

```diff
@@ -113,3 +113,29 @@ func PublicIPs() ([]string, error) {
 
 	return ips, nil
 }
+
+// InstanceIP checks if any of the IP's in input IP list belong to the given network CIDR.
+// It returns the first matching IP or an error.
+func InstanceIP(ips []string, networkCIDR string) (net.IP, error) {
```

Comment:
How would you use this?
```
all, err := PrivateIPs()
if err != nil {
    ...
}
ip, err := InstanceIP(all, "10.0.0.0/24")
if err != nil {
    ...
}
```

Consider this instead:
```
ip, err := PrivateIPs("10.0.0.0/24")
if err != nil {
    ...
}
```

What was the reasoning ?

Reply:
> How would you use this? 

Yes, if you look at the tomato PR... this is exactly what I was doing...

```go
	ips, err := florist.PrivateIPs()
	if err != nil {
		return nil, err
	}
	instanceIP, err := florist.InstanceIP(ips, VpcSubnet)
	if err != nil {
		return nil, err
	}
```

> What was the reasoning ?

You are correct... Let me simplify...